src/platform_impl/macos/event.rs

```diff
@@ -29,216 +50,265 @@ pub enum EventProxy {
-        '3' | '#' => VirtualKeyCode::Key3,
-        '4' | '$' => VirtualKeyCode::Key4,
-        '5' | '%' => VirtualKeyCode::Key5,
-        '6' | '^' => VirtualKeyCode::Key6,
-        '7' | '&' => VirtualKeyCode::Key7,
-        '8' | '*' => VirtualKeyCode::Key8,
-        '9' | '(' => VirtualKeyCode::Key9,
-        '0' | ')' => VirtualKeyCode::Key0,
-        '=' | '+' => VirtualKeyCode::Equals,
-        '-' | '_' => VirtualKeyCode::Minus,
-        ']' | '}' => VirtualKeyCode::RBracket,
-        '[' | '{' => VirtualKeyCode::LBracket,
-        '\'' | '"' => VirtualKeyCode::Apostrophe,
-        ';' | ':' => VirtualKeyCode::Semicolon,
-        '\\' | '|' => VirtualKeyCode::Backslash,
-        ',' | '<' => VirtualKeyCode::Comma,
-        '/' | '?' => VirtualKeyCode::Slash,
-        '.' | '>' => VirtualKeyCode::Period,
-        '`' | '~' => VirtualKeyCode::Grave,
-        _ => return None,
-    })
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct KeyEventExtra {
+    pub text_with_all_modifiers: Option<&'static str>,
+    pub key_without_modifiers: Key<'static>,
+}
+
+impl KeyEventExtModifierSupplement for KeyEvent {
+    fn text_with_all_modifiers(&self) -> Option<&str> {
+        self.platform_specific.text_with_all_modifiers
+    }
+
+    fn key_without_modifiers(&self) -> Key<'static> {
+        self.platform_specific.key_without_modifiers
+    }
+}
+
+pub fn get_modifierless_char(scancode: u16) -> Key<'static> {
+    let mut string = [0; 16];
+    let input_source;
+    let layout;
+    unsafe {
+        input_source = ffi::TISCopyCurrentKeyboardLayoutInputSource();
+        if input_source.is_null() {
+            log::error!("`TISCopyCurrentKeyboardLayoutInputSource` returned null ptr");
+            return Key::Unidentified(NativeKeyCode::MacOS(scancode));
+        }
+        let layout_data =
+            ffi::TISGetInputSourceProperty(input_source, ffi::kTISPropertyUnicodeKeyLayoutData);
+        if layout_data.is_null() {
+            CFRelease(input_source as *mut c_void);
+            log::error!("`TISGetInputSourceProperty` returned null ptr");
+            return Key::Unidentified(NativeKeyCode::MacOS(scancode));
+        }
+        layout = CFDataGetBytePtr(layout_data) as *const ffi::UCKeyboardLayout;
+    }
+    let keyboard_type = unsafe { ffi::LMGetKbdType() };
+
+    let mut result_len = 0;
+    let mut dead_keys = 0;
+    let modifiers = 0;
+    let translate_result = unsafe {
+        ffi::UCKeyTranslate(
+            layout,
+            scancode,
+            ffi::kUCKeyActionDisplay,
+            modifiers,
+            keyboard_type as u32,
+            ffi::kUCKeyTranslateNoDeadKeysMask,
+            &mut dead_keys,
+            string.len() as ffi::UniCharCount,
+            &mut result_len,
+            string.as_mut_ptr(),
+        )
+    };
+    unsafe {
+        CFRelease(input_source as *mut c_void);
+    }
+    if translate_result != 0 {
+        log::error!("`UCKeyTranslate` returned 0");
```

Comment:
Should this be `returned non-zero`

Reply:
You are absolutely correct sir. I addressed this in 579b257e15363c58f6f9ae60af1931bb35383853